### PR TITLE
Hardware pwm

### PR DESCRIPTION
**Pull request by mck1117**
On pins that support it, we can use hardware PWM instead of software.  This relieves scheduler pressure, and improves PWM accuracy.

### CODE REVIEW

**Comment by mck1117** [comment]
As an added bonus, hardware PWM stays running during flash erase/write... #1261

**Comment by mck1117** [comment]
Tested my ETB up to ~10khz and down to 50hz PWM - works great!

**Comment by mck1117** [comment]
The pins configured for hardware PWM with this PR are PB6-9, PC6-9, and PD12-15.  Most importantly that covers the ETB PWM pins on both MRE and Proteus.

**Comment by rusefi** [comment]
@mck1117  poke?

File: firmware/hw_layer/ports/stm32/stm32_common.cpp [hunk 1]
```diff
@@ -252,6 +252,11 @@ stm32_hardware_pwm pwmChannels[] = {
 
 /*static*/ hardware_pwm* hardware_pwm::tryInitPin(const char* msg, brain_pin_e pin, float frequencyHz, float duty)
 {
+	// Slow PWM isn't worth doing on hardware - reserve that timer for something faster if needed.
+	if (frequencyHz < 100) {
```
Comment (rusefi) [inline firmware/hw_layer/ports/stm32/stm32_common.cpp:256]: that's really weird to be choosing HW at runtime. Does not look right to me.

Comment (mck1117) [reply firmware/hw_layer/ports/stm32/stm32_common.cpp:256]: You mean like how we already do this for ICU inputs, EXTI inputs, SPI devices, CAN devices, serial devices, etc?

Comment (rusefi) [reply firmware/hw_layer/ports/stm32/stm32_common.cpp:256]: Looks a bit different to me - in ICU inputs, EXTI inputs, SPI devices, CAN devices, serial devices we select HW _directly_ based on _HW_ parameters

here we choose HW based on something not obviously connected to HW.

Comment (mck1117) [reply firmware/hw_layer/ports/stm32/stm32_common.cpp]: comment & threshold updated to the actual limit of what this implementation can do

File: firmware/hw_layer/ports/kinetis/mpu_util.cpp [hunk 2]
```diff
@@ -257,5 +257,10 @@ uintptr_t getFlashAddrSecondCopy() {
 	return 0x10008000;
 }
 
+/*static*/ hardware_pwm* hardware_pwm::tryInitPin(const char*, brain_pin_e, float, float) {
+	// TODO: implement me!
```
Comment (rusefi) [inline firmware/hw_layer/ports/kinetis/mpu_util.cpp:261]: what is the plan here?

Comment (mck1117) [reply firmware/hw_layer/ports/kinetis/mpu_util.cpp:261]: This isn't required to be implemented, it just means hardware PWM won't ever be used on Kinetis.

**Pull request merged**